build-scripts/build_tests.py

```diff
@@ -0,0 +1,230 @@
+    )
+    parser.add_argument(
+        "--resolved-rules-dir", required=True,
+        help="Directory with <rule-id>.yml resolved rule YAMLs"
+             "e.g.: ~/scap-security-guide/build/rhel10/rules"
+    )
+    parser.add_argument("--log-level", action="store", type=str, default="ERROR",
+                        choices=["ERROR", "WARNING", "INFO", "DEBUG", "TRACE"],
+                        help="What level to log at. Defaults to ERROR.")
+    parser.add_argument("--root", default=SSG_ROOT,
+                        help=f"Path to the project. Defaults to {SSG_ROOT}")
+    parser.add_argument("--jobs", "-j", type=int, default=JOB_COUNT,
+                        help=f"Number of cores to use. Defaults to {JOB_COUNT} on this system.")
+    return parser
+
+
+def _write_path(file_contents: str, output_path: os.PathLike) -> None:
+    with open(output_path, "w") as file:
+        file.write(file_contents)
+        file.write("\n")
+
+
+def _is_test_file(filename: str) -> bool:
+    return filename.endswith(('.pass.sh', '.fail.sh', '.notapplicable.sh'))
+
+
+def _get_deny_templated_scenarios(test_config_path: pathlib.Path) -> Set[str]:
+    if test_config_path.exists():
+        test_config = ssg.yaml.open_raw(str(test_config_path.absolute()))
+        deny_templated_scenarios = test_config.get('deny_templated_scenarios', set())
+        return deny_templated_scenarios
+    return set()
+
+
+def _process_shared_file(env_yaml: dict, file: pathlib.Path, shared_output_path: pathlib.Path) \
+        -> None:
+    file_contents = ssg.jinja.process_file_with_macros(str(file.absolute()), env_yaml)
+    shared_script_path = shared_output_path / file.name
+    _write_path(file_contents, shared_script_path)
+
+
+def _copy_and_process_shared(env_yaml: dict, output_path: pathlib.Path, root_path: pathlib.Path) \
+        -> None:
+    tests_shared_root = root_path / "tests" / "shared"
+    shared_output_path = output_path / "shared"
+    shared_output_path.mkdir(parents=True, exist_ok=True)
+    for file in tests_shared_root.iterdir():  # type: pathlib.Path
+        # We only support one level deep, this avoids recursive functions
+        if file.is_dir():
+            for sub_file in file.iterdir():
+                shared_output_path_sub = shared_output_path / file.name
+                shared_output_path_sub.mkdir(parents=True, exist_ok=True)
+                _process_shared_file(env_yaml, sub_file, shared_output_path_sub)
+        else:
+            _process_shared_file(env_yaml, file, shared_output_path)
+
+
+def _process_local_tests(benchmark_cpes: Set[str], env_yaml: dict, rule_output_path: pathlib.Path,
+                         rule_tests_root: pathlib.Path) -> None:
+    logger = logging.getLogger()
+    for test in rule_tests_root.iterdir():  # type: pathlib.Path
+        if not test.name.endswith(".sh"):
+            logger.debug("Skipping file %s in rule %s is it doesn't end with .sh.",
```

Comment:
This will skip files like for example `linux_os/guide/auditing/auditd_configure_rules/audit_file_modification/audit_rules_unsuccessful_file_modification/tests/test_audit.rules` which are need for the test scenarios to work.